channeldb/db.go

```diff
@@ -556,70 +556,102 @@ func (d *DB) FetchChannel(chanPoint wire.OutPoint) (*OpenChannel, error) {
+}
+
+// fetchChannelsFilter applies a filter to channels retrieved in fetchchannels.
+// Filters are combined to produce
+type fetchChannelsFilter struct {
+	// include is a function that returns true if a channel meets our
+	// search criteria.
+	include func(channel *OpenChannel) bool
+
+	// description is a string which explains which channels are being
+	// included.
+	description string
+}
+
+// pendingChannelFilter returns a filter based on whether channels are pending
+// (ie, their funding transaction still needs to confirm). If pending is false,
+// channels with confirmed funding transactions are returned.
+var pendingChannelFilter = func(pending bool) fetchChannelsFilter {
+	return fetchChannelsFilter{
+		include: func(channel *OpenChannel) bool {
+			return channel.IsPending == pending
+		},
+		description: fmt.Sprintf("fetching channels with "+
+			"pending: %v", pending),
 	}
+}
 
-	return append(waitingClose, pendingWaitingClose...), nil
+// waitingCloseFilter returns a filter which filters channels based on whether
+// they are awaiting the confirmation of their closing transaction. If waiting
+// close is true, channels that have had their closing tx broadcast are
+// included. If it is false, channels that are not awaiting confirmation of
+// their close transaction are returned.
+var waitingCloseFilter = func(waitingClose bool) fetchChannelsFilter {
+	return fetchChannelsFilter{
+		include: func(channel *OpenChannel) bool {
+			// If the channel is in any other state than Default,
+			// then it means it is waiting to be closed.
+			channelWaitingClose :=
+				channel.ChanStatus() != ChanStatusDefault
+
+			// Include the channel if it matches the value for
+			// waiting close that we are filtering on.
+			return channelWaitingClose == waitingClose
+		},
+		description: fmt.Sprintf("fetching channels with "+
+			"waiting close: %v", waitingClose),
+	}
 }
 
 // fetchChannels attempts to retrieve channels currently stored in the
-// database. The pending parameter determines whether only pending channels
-// will be returned, or only open channels will be returned. The waitingClose
-// parameter determines whether only channels waiting for a closing transaction
-// to be confirmed should be returned. If no active channels exist within the
-// network, then ErrNoActiveChannels is returned.
-func fetchChannels(d *DB, pending, waitingClose bool) ([]*OpenChannel, error) {
+// database. It takes a set of filters which are applied to each channel to
+// obtain a set of channels with the desired set of properties. Only channels
+// which have a true value returned for *all* of the filters will be returned.
+// If no filters are provided, every channel in the open channels bucket will
+// be returned.
+func fetchChannels(d *DB, filters ...fetchChannelsFilter) ([]*OpenChannel, error) {
 	var channels []*OpenChannel
 
+	// Log the number of filters and their descriptions.
+	log.Infof("fetchChannels applying: %v filters", len(filters))
```

Comment:
Would it be better to have this as a debug log? Do we frequently use this information in logs?

Reply:
Could be more of a trace log, or something not even worth logging possibly.

Reply:
Just going to kill the logging, not much value add and can be quite spammy.